pkg/controller/jobs/job/job_multikueue_adapter.go

```diff
@@ -49,6 +50,12 @@ func (b *multikueueAdapter) SyncJob(ctx context.Context, localClient client.Clie
 		return err
 	}
 
+	log := ctrl.LoggerFrom(ctx)
+	if ptr.Deref(localJob.Spec.Suspend, false) { // Ensure the job is resumed before updating its status; otherwise, it will fail when patching the spec.
```

Comment:
This changes the implementation - is this actually a bug? If so, we should add a release note describing the scenario that is fixed

Reply:
Could you maybe explain a bit more what is the update that is failing trying to update in the spec?

Reply:
Ah I think I know what you mean, k8s core blocks update of pod template if the job is running. And it determines that the job is running because status.startTime Is not bil

Reply:
FYI: [here](https://github.com/kubernetes/kubernetes/blob/810e9e212ec5372d16b655f57b9231d8654a2179/pkg/registry/batch/job/strategy.go#L192-L194) and [here](https://github.com/kubernetes/kubernetes/blob/810e9e212ec5372d16b655f57b9231d8654a2179/pkg/apis/batch/validation/validation.go#L641-L662) are the relevant places in the core k8s which reject the spec update if the status is already updated (status.startTime != nil).